### PR TITLE
Fix TestPushWriteRequest for remotewriteprotocol for pure-test (with native zstd)

### PR DESCRIPTION
**Pull request by Amper**
Add tolerance interval to comparing block size for VictoriaMetrics remote write in TestPushWriteRequest

### CODE REVIEW

File: app/vmagent/remotewrite/pendingseries_test.go
Comment (hagen1778) [inline app/vmagent/remotewrite/pendingseries_test.go:40]: nit: just to simplify things

```suggestion
if pushBlockLen == expectedBlockLen {
	return
}

if tolerancePrc == 0 {
	t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes",
	rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen)
}

if math.Abs(float64(pushBlockLen-expectedBlockLen)/float64(expectedBlockLen)*100) > tolerancePrc {
	t.Fatalf("unexpected block len for rowsCount=%d, isVMRemoteWrite=%v; got %d bytes; expecting %d bytes +- %.0f%%",
	rowsCount, isVMRemoteWrite, pushBlockLen, expectedBlockLen, tolerancePrc)
}
```

**Review by hagen1778 — COMMENTED** [review]
Would be nice to explain in more detail why we're having this change in the commit message.
Although, I don't know is 15% of tolerance is something acceptable.

File: app/vmagent/remotewrite/pendingseries_test.go
Comment (Amper) [reply app/vmagent/remotewrite/pendingseries_test.go]: > Would be nice to explain in more detail why we're having this change in the commit message.
> Although, I don't know is 15% of tolerance is something acceptable.

To tell the truth, i am not at all sure about the need to check the size of the compressed message in tests, just because the compression algorithm is constantly change and theoretically may depend on the environment in which it is launched.
Now the value of is selected empirically, such a solution was proposed yesterday by @valyala

Comment (hagen1778) [reply app/vmagent/remotewrite/pendingseries_test.go]: Ok, thanks! Let's wait for a review from @valyala or @f41gh7, then.

Comment (f41gh7) [reply app/vmagent/remotewrite/pendingseries_test.go]: Any way it'll be flaky test and checking pushBlockLen doesn't make much sense, as Amper mentioned compression rate may be different.

 Let's merge it as is.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Review by hagen1778 — APPROVED** [review]

**Comment by hagen1778** [comment]
Hey @Amper, could you please configure commits signing ? See more at https://docs.github.com/articles/about-gpg/

**Pull request merged**
